source/compiler/qsc_parse/src/lex/cooked/tests.rs

```diff
@@ -679,6 +680,86 @@ fn leading_zero_float() {
     );
 }
 
+#[test]
+fn trailing_i_complex() {
+    check(
+        "4.2i",
+        &expect![[r#"
+            [
+                Ok(
+                    Token {
+                        kind: Imaginary,
+                        span: Span {
+                            lo: 0,
+                            hi: 4,
+                        },
+                    },
+                ),
+            ]
+        "#]],
+    );
+}
+
+#[test]
+fn trailing_dot_trailing_i_complex() {
+    check(
+        "4.i",
+        &expect![[r#"
+            [
+                Ok(
+                    Token {
+                        kind: Imaginary,
+                        span: Span {
+                            lo: 0,
+                            hi: 3,
+                        },
+                    },
+                ),
+            ]
+        "#]],
+    );
+}
+
+#[test]
+fn leading_dot_trailing_i_int_ident() {
+    check(
+        ".1i",
```

Comment:
This test shows that this didn't make an imaginary token. That seems wrong, no?

Reply:
While we do support leaving off the trailing zero for a double, as in `1.`, we do not support leaving of the leading zero. So `.1` from a parsing perspective is not a Double literal but rather a "dot" token followed by an Int literal. We have other unit tests in the file that confirm this behavior, I just wanted to make sure the interaction between that and the new literal `i` syntax produced the expected token stream, namely "dot" followed by Int literal followed by ident "i" tokens.

Reply:
That makes sense. Thanks!